sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientAssertionCredential.java

```diff
@@ -119,8 +119,10 @@ public Mono<AccessToken> getToken(TokenRequestContext request) {
     public AccessToken getTokenSync(TokenRequestContext request) {
         try {
             AccessToken token = identitySyncClient.authenticateWithConfidentialClientCache(request);
-            LoggingUtil.logTokenSuccess(LOGGER, request);
-            return token;
+            if (token != null) {
+                LoggingUtil.logTokenSuccess(LOGGER, request);
+                return token;
+            }
```

Comment:
Aside from the logging change, was this a bug before if a null token was returned here that we didn't continue to the next possible token retrieval call?

Reply:
No. It either returned a token or threw an exception. I'm adding the null state to represent the cache miss.